examples/platform/silabs/wifi/icd/ApplicationSleepManager.h

```diff
@@ -131,20 +133,54 @@ class ApplicationSleepManager : public chip::app::ReadHandler::ApplicationCallba
 
     void OnFabricUpdated(const chip::FabricTable & fabricTable, chip::FabricIndex fabricIndex) override {}
 
+    // ICDStateObserver implementation
+    void OnEnterActiveMode() override;
+    void OnEnterIdleMode() override;
+    void OnTransitionToIdle() override;
+    void OnICDModeChange() override;
+
 private:
     ApplicationSleepManager()  = default;
     ~ApplicationSleepManager() = default;
 
     ApplicationSleepManager(const ApplicationSleepManager &)             = delete;
     ApplicationSleepManager & operator=(const ApplicationSleepManager &) = delete;
 
+    /**
+     * @brief Processes special cases based on the vendor ID.
+     *
+     * This method checks if the given vendor ID has any special cases that allow
+     * the device to go to LI based sleep when the fabric associated to the vendor ID does not have an active subscription.
+     *
+     * @param vendorId The vendor ID to check for special cases.
+     * @return true if the vendor ID has a special case that allows LI based sleep, false otherwise.
+     */
+    bool ProcessSpecialVendorIDCase(chip::VendorId vendorId);
+
+    /**
+     * @brief Processes the Apple Keychain edge case.
+     *
+     * Apple, when commissioning, adds two fabric to the device. One for Apple Home and one for the Appley Keychain.
+     * Apple Home is the active fabric which is used to communication with the device. The associated fabric also has the active
+     * subcription. Applye Keychain fabric acts as a safety and doesn't have an active fabric with the device. As such, we need an
+     * alternate method to check if the device can go to LI based sleep.
+     *
+     * This method checks if there is any fabric with the Apple Home vendor ID that
+     * has at least one active subscription. If such a fabric is found, it allows
+     * the device to go to LI based sleep.
+     *
+     * @return true if the Apple Keychain edge case allows low-power mode, false otherwise.
+     */
+    bool ProcessKeychainEdgeCase();
```

Comment:
I know this is very explicit by the comment above, but how about putting _Apple_ in the method name so when we se it called in the code we don't have to come back here to know what is going on?

(Or we could have a functioning intellisense that shows the comment when we hover the function but who has this kind of technology...)

Reply:
I made the inverse comment before (he had apple in) IMO we should not publically mention Apple in our function names